app/models/utils/generators/main_generator.rb

```
@@ -24,10 +24,13 @@ def generate
         end
         ActionMailer::Base.perform_deliveries = true
         cluster
-      rescue StandardError
+      rescue StandardError => ex
         # Can't create the cluster inside the transaction (see above). So we need to clean up in here instead
         # in case of error.
-        cluster.destroy
+        ActsAsTenant.with_tenant(cluster) do
+          cluster.destroy
+        end
+        raise ex
```

This error was getting swallowed, so I wanted to make sure it got resurfaced